### PR TITLE
chore(compass-saved-aggregations-queries): validate loaded saved pipelines to make sure that the schema matches COMPASS-5448

### PR DESCRIPTION
**Pull request by gribnoysup**
The original ticket was only about corrupted files, but Rhys found another corner case: by removing specific keys from the file you could still break displaying all aggregation items. This patch adds some additional validation to the pipeline store that checks that the loaded document has all the required keys even if we managed to load it successfully

### CODE REVIEW

File: packages/compass-saved-aggregations-queries/src/stores/copy-to-clipboard.ts [hunk 1]
```diff
@@ -75,7 +47,7 @@ export function copyToClipboard(
 
     const copyStr =
       item.type === 'aggregation'
-        ? formatPipeline(item.aggregation.pipeline)
+        ? item.aggregation.pipelineText
```
Comment (gribnoysup) [inline packages/compass-saved-aggregations-queries/src/stores/copy-to-clipboard.ts:50]: Drive-by: we forgot to adjust for the changes in the stored pipeline format, no need to build the string here anymore, we can just return `pipelineText` directly

**Review by Anemy — APPROVED** [review]
👍

File: packages/compass-aggregations/package.json [hunk 2]
```diff
@@ -5,17 +5,19 @@
...
     "require": "./dist/index.js"
   },
   "compass:exports": {
-    ".": "./src/index.js"
+    ".": "./src/index.ts"
   },
   "scripts": {
     "prepublishOnly": "npm run compile && compass-scripts check-exports-exist",
     "compile": "npm run webpack -- --mode production",
+    "postcompile": "tsc --emitDeclarationOnly",
```
[the remark this answers is not in the document]
Comment (gribnoysup) [reply packages/compass-aggregations/package.json:20]: This still requires to recompile manually when working on multiple packages unfortunately which I really want to solve at some point, but should do it for distribution
